app/assets/javascripts/discourse/app/components/discovery/filter-tips.gjs

```diff
@@ -0,0 +1,552 @@
+      });
+
+      // If the tip has a type and we have value text, do placeholder search
+      if (tip?.type && valueText !== undefined) {
+        this.handlePlaceholderSearch(filterName, valueText, tip, prefix);
+        return this.searchResults.length > 0 ? this.searchResults : [];
+      }
+    }
+
+    // This handles blank, the default state when nothing is typed
+    if (!this.currentInputValue || lastWord === "") {
+      return this.args.tips
+        .filter((tip) => tip.priority)
+        .sort((a, b) => (b.priority || 0) - (a.priority || 0))
+        .slice(0, MAX_RESULTS);
+    }
+
+    const tips = [];
+
+    this.args.tips.forEach((tip) => {
+      if (tips.length >= MAX_RESULTS) {
+        return;
+      }
+      const tipName = tip.name;
+      const searchTerm = lastWord.substring(prefix.length);
+
+      // Skip exact matches with colon
+      if (searchTerm.endsWith(":") && tipName === searchTerm) {
+        return;
+      }
+
+      const prefixMatch =
+        searchTerm === "" &&
+        prefix &&
+        tipName.prefixes &&
+        tipName.prefixes.find((p) => p.name === prefix);
+
+      if (prefixMatch || tipName.indexOf(searchTerm) > -1) {
+        this.pushPrefixTips(tip, tips, null, prefix);
+        if (!prefix) {
+          tips.push(tip);
+        }
+      } else if (tip.alias && tip.alias.indexOf(searchTerm) > -1) {
+        this.pushPrefixTips(tip, tips, tip.alias, prefix);
+        tips.push({
+          ...tip,
+          name: tip.alias,
+        });
+      }
+    });
+
+    return tips.sort((a, b) => {
+      const aName = a.name;
+      const bName = b.name;
+      const searchTerm = lastWord;
+
+      const aStartsWith = aName.startsWith(searchTerm);
+      const bStartsWith = bName.startsWith(searchTerm);
+
+      if (aStartsWith && !bStartsWith) {
+        return -1;
+      }
+      if (!aStartsWith && bStartsWith) {
+        return 1;
+      }
+
+      if (aStartsWith && bStartsWith) {
+        if (aName.length !== bName.length) {
+          return aName.length - bName.length;
+        }
+      }
+
+      return aName.localeCompare(bName);
+    });
+  }
+
+  pushPrefixTips(tip, tips, alias = null, currentPrefix = null) {
+    if (tip.prefixes && tip.prefixes.length > 0) {
+      tip.prefixes.forEach((prefix) => {
+        if (currentPrefix && !prefix.name.startsWith(currentPrefix)) {
+          return;
+        }
+        tips.push({
+          ...tip,
+          name: `${prefix.name}${alias || tip.name}`,
+          description: prefix.description || tip.description,
+          isPlaceholderCompletion: true,
+        });
+      });
+    }
+  }
+
+  extractPrefix(word) {
+    const match = word.match(/^(-=|=-|-|=)/);
+    return match ? match[0] : "";
+  }
+
+  @action
+  handlePlaceholderSearch(filterName, valueText, tip, prefix = "") {
+    this.activeFilter = filterName;
+
+    if (this.searchTimer) {
+      cancel(this.searchTimer);
+    }
+
+    this.searchTimer = debounce(
```

Comment:
It seems like much more often we use `discourseDebounce` which uses ember debounce internally in the codebase:

```
import discourseDebounce from "discourse/lib/debounce";
```

Works in the same way but handles our qunit tests